cpp/include/cudf/io/parquet.hpp

```diff
@@ -389,9 +389,10 @@ class column_in_metadata {
   bool _use_int96_timestamp = false;
   // bool _output_as_binary = false;
   thrust::optional<uint8_t> _decimal_precision;
-  std::vector<column_in_metadata> children;
 
  public:
+  std::vector<column_in_metadata> children;
```

Comment:
No other data members are in this class are public. Instead, they have getters and setters. Why did you make `children` public?

Reply:
I am not a cpp expert so please let me know if there is a better way to do this. 

The way @devavret has designed the API works perfectly when we have a table and we initialize the table_input_metadata with it, if we don't then the children vector is never initialized this is why I made it public so I can initialize it.

Reply:
You are violating encapsulation. @devavret or @vuule who know these classes better should review.

Reply:
Thanks, I have spoken to @devavret about this but I would love for him or @vuule to review.

Reply:
This was allowed for the time being to unblock the API development. It's how the old metadata class functioned. I'm thinking now whether adding a `add_column()` method would be useful. It would still need to construct its entire hierarchy at once.

Reply:
At least add a comment. This really should be done the right way, it wouldn't be hard to add a `get_children` accessor...

Reply:
@harrism I have added a getter for the private member.